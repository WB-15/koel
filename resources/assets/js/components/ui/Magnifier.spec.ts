import { expect, it } from 'vitest'
import UnitTestCase from '@/__tests__/UnitTestCase'
import { fireEvent } from '@testing-library/vue'
import Magnifier from './Magnifier.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders and functions', async () => {
      const { getByTitle, html, emitted } = this.render(Magnifier)

      await fireEvent.click(getByTitle('Zoom in'))
      expect(emitted()['in']).toBeTruthy()

      await fireEvent.click(getByTitle('Zoom out'))
      expect(emitted().out).toBeTruthy()

      expect(html()).toMatchSnapshot()
    })
  }
}
