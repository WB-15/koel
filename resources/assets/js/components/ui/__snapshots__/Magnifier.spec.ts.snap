// Vitest Snapshot v1

exports[`renders and functions 1`] = `<span data-v-fcc3eddd=""><button data-v-fcc3eddd="" title="Zoom out" type="button"><br data-v-fcc3eddd="" data-testid="icon" icon="[object Object]"></button><button data-v-fcc3eddd="" title="Zoom in" type="button"><br data-v-fcc3eddd="" data-testid="icon" icon="[object Object]"></button></span>`;
